mods/cnc/maps/nod08a/nod08a.lua

```diff
@@ -0,0 +1,223 @@
+	player.MarkCompletedObjective(NodObjective1)
+	Utils.Do(NodBase, function(actor)
+		actor.Owner = player
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Media.PlaySpeechNotification(player, "NewOptions")
+	end)
+end
+
+searches = 0
+getAirstrikeTarget = function()
+	local list = player.GetGroundAttackers()
+
+	if #list == 0 then
+		return
+	end
+
+	local target = list[DateTime.GameTime % #list + 1].CenterPosition
+
+	if searches < 6 then
+		searches = searches + 1
+		return getAirstrikeTarget()
+	else
+		searches = 0
+		return target
+	end
+end
+
+--Provide the player with a helicopter until the outpost got captured
+SendHelicopter = function()
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		if not outpostCaptured then
+			Media.PlaySpeechNotification(player, "Reinforce")
+			TransportHelicopter = Reinforcements.ReinforceWithTransport(player, 'tran', nil, { ReinforcementsHelicopterSpawn.Location, waypoint15.Location })[1]
+			Trigger.OnKilled(TransportHelicopter, function()
+					SendHelicopter()
+			end)
+		end
+	end)
+end
+
+SendAttackWave = function(team)
+	for type, amount in pairs(team.units) do
+		count = 0
+		actors =  enemy.GetActorsByType(type)
+		Utils.Do(actors, function(actor)
+			if actor.IsIdle and count < amount then
+				SetAttackWaypoints(actor, team.waypoints)
+				IdleHunt(actor)
+				count = count + 1
+			end
+		end)
+	end
+end
+
+SetAttackWaypoints = function(actor, waypoints)
+	if not actor.IsDead then
+		Utils.Do(waypoints, function(waypoint)
+			actor.AttackMove(waypoint.Location)
+		end)
+	end
+end
+
+SendGDIAirstrike = function(hq, delay)
+	if not hq.IsDead and hq.Owner == enemy then
+		local target = getAirstrikeTarget()
+
+		if target then
+			hq.SendAirstrike(target, false, Facing.NorthEast + 4)
+			Trigger.AfterDelay(delay, function() SendGDIAirstrike(hq, delay)  end)
+		else
+			Trigger.AfterDelay(delay/4, function() SendGDIAirstrike(hq, delay)  end)
+		end
+	end
+end
+
+SendWaves = function(counter, Waves)
+	if counter <= #Waves then
+		team = Waves[counter]
+		SendAttackWave(team)
+		Trigger.AfterDelay(DateTime.Seconds(team.delay), function() SendWaves(counter + 1, Waves) end)
+	end
+end
+
+StartWaves = function()
+	SendWaves(1, AutoAttackWaves)
+end
+
+Trigger.OnAllKilled(IntroGuards, function()
+	FlareCamera = Actor.Create("camera", true, { Owner = player, Location = waypoint25.Location })
+	Flare = Actor.Create("flare", true, { Owner = player, Location = waypoint25.Location })
+	SendHelicopter()
+	player.MarkCompletedObjective(NodObjective1)
+	NodBaseCapture()
+end)
+
+Trigger.OnAllKilledOrCaptured(Outpost, function()
+	if not outpostCaptured then
+		outpostCaptured = true
+
+		Trigger.AfterDelay(DateTime.Minutes(1), function()
+			airstrikeproxy = Actor.Create("airstrike.proxy", false, { Owner = enemy })
+
+			if not GDIHQ.IsDead and (not NodHand.IsDead or not NodNuke.IsDead) then
+				airstrikeproxy.SendAirstrike(AirstrikeTarget.CenterPosition, false, Facing.NorthEast + 4)
```

Comment:
Can we move the actor creation etc into the check as well?
```
if not GDIHQ.IsDead and (not NodHand.IsDead or not NodNuke.IsDead) then 
	local airstrikeproxy = Actor.Create("airstrike.proxy", false, { Owner = enemy }) 
	airstrikeproxy.SendAirstrike(AirstrikeTarget.CenterPosition, false, Facing.NorthEast + 4) 
	airstrikeproxy.Destroy() 
end
```
Also notice the `local` keyword. As we don't want to reference the airstrike actor from elsewhere again
I think we don't need to have this global.